### ISSUE TITLE
az network asg create throws an error

### ISSUE THREAD

**Issue by mbachmakov** [Azure/azure-cli#23604]
## Describe the bug

**Command Name**
`az network asg create`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'MSIAuthenticationWrapper' object has no attribute 'get_auxiliary_tokens'
Traceback (most recent call last):
  File "/opt/az/lib/python3.10/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 708, in _run_job
    result = LongRunningOperation(cmd_copy.cli_ctx, 'Starting {}'.format(cmd_copy.name))(result)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 1000, in __call__
    result = poller.result()
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_poller.py", line 105, in result
    self.wait(timeout)
  File "/opt/az/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 73, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_poller.py", line 127, in wait
    raise self._exception  # type: ignore
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_poller.py", line 80, in _start
    for polling_method in self._polling_generator:
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/network/aaz/latest/network/asg/_create.py", line 73, in _execute_operations
    yield self.ApplicationSecurityGroupsCreateOrUpdate(ctx=self.ctx)()
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/network/aaz/latest/network/asg/_create.py", line 84, in __call__
    session = self.client.send_request(request=request, stream=False, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_client.py", line 86, in send_request
    session = self._pipeline.run(request, stream=stream, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/_base.py", line 211, in run
    return first_node.send(pipeline_request)  # type: ignore
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/_base.py", line 71, in send
    response = self.next.send(request)
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/_base.py", line 71, in send
    response = self.next.send(request)
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/_base.py", line 71, in send
    response = self.next.send(request)
  [Previous line repeated 2 more times]
  File "/opt/az/lib/python3.10/site-packages/azure/mgmt/core/policies/_base.py", line 47, in send
    response = self.next.send(request)
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/policies/_redirect.py", line 158, in send
    response = self.next.send(request)
  File "/opt/az/lib/python3.10/site-packages/azure/core/pipeline/policies/_retry.py", line 446, in send
    response = self.next.send(request)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_http_policy.py", line 105, in send
    self.on_request(request)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/aaz/_http_policy.py", line 81, in on_request
    self._aux_tokens = self._credential.get_auxiliary_tokens(*self._scopes)
AttributeError: 'MSIAuthenticationWrapper' object has no attribute 'get_auxiliary_tokens'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az network asg create --resource-group {} --name {}`

## Environment Summary
```
Linux-5.4.0-1089-azure-x86_64-with-glibc2.28 (Cloud Shell), Common Base Linux Delridge (quinault)
Python 3.10.5
Installer: DEB

azure-cli 2.39.0

Extensions:
ai-examples 0.2.5
ssh 1.1.2

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
network

**Comment by necusjz** [comment]
It will be fixed in Azure CLI 2.40.0

**Comment by jsntcy** [comment]
Fixed in https://github.com/Azure/azure-cli/pull/23503
